### PR TITLE
Allow self as discovery responder.

### LINKED ISSUE #723: Disco Mode: Let users know that the only responder is themselves
We made a decision earlier on to show "unable to initiate session" even if the only responder is the Hotspot itself. This creating some confusion so we should let them know that everything worked, but no one else heard them.

So instead of showing the error, we should show them the result with the Hotspot as a responder.

### PR DESCRIPTION
**Pull request by matthewcarlreetz**
Closes #723

### CODE REVIEW

File: src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { DiscoveryResponse } from '../../../../store/discovery/discoveryTypes'
+
+export default (responses: DiscoveryResponse[]) =>
+  responses.filter(
```
Comment (matthewcarlreetz) [inline src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts:4]: This is the bulk of the change. We used to filter out "self" here.

**Review by tyler-whitman — APPROVED** [review]

File: src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts (hunk 1 above)
Comment (allenan) [reply src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts:4]: would [lodash/uniqBy](https://lodash.com/docs/4.17.15#uniqBy) be useful here? sometimes they use clever algorithms under the hook for these utilities -- was just thinking that filter + findIndex might create a few loops through the set.

Comment (matthewcarlreetz) [reply src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts:4]: Yeah, I think so. If nothing else, it makes the code more readable.

File: src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { uniqBy } from 'lodash'
+import { DiscoveryResponse } from '../../../../store/discovery/discoveryTypes'
+
+export default (responses: DiscoveryResponse[]) =>
+  uniqBy(responses, (response) => response.hotspotAddress)
```
Comment (allenan) [inline src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts:5]: i think you can also do this like `uniqBy(responses, 'hotspotAddress')`

Comment (matthewcarlreetz) [reply src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts:5]: oh, cool. I've never used it like that before.

**Review by allenan — APPROVED** [review]
looks good. one little lodash syntax suggestion above ^

File: src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts (hunk 2 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/settings/discovery/dedupeDiscoveryResponses.ts:5]: For TS I think I prefer `uniqBy(responses, ({ hotspotAddress }) => hotspotAddress)`
That way we don't lose type safety.

**Pull request merged**